node/src/services/builder.rs

```diff
@@ -98,105 +163,170 @@ where
         self
     }
 
-    /// Add a new [`FileStorage`] to the builder.
-    ///
-    /// This is the set of tools that allows a StorageHub node to store files as Merkle Patricia
-    /// Tries, in the way that the StorageHub protocol specifies.
-    pub fn with_file_storage(&mut self, file_storage: Arc<RwLock<FL>>) -> &mut Self {
-        self.file_storage = Some(file_storage);
-        self
+    fn build_handler(
```

Comment:
*(nitpick)*

This could also be typed so that the compiler doesn't let you build something invalid or without missing parts.